apps/dot_ext/tests/test_verify_bfd_headers.py

```diff
@@ -0,0 +1,84 @@
+            'client_id': application.client_id,
+            'response_type': 'code',
+            'redirect_uri': application.redirect_uris,
+            'scope': application.scopes().split(" "),
+            'expires_in': 86400,
+            'allow': True,
+        }
+        if application.authorization_grant_type == Application.GRANT_IMPLICIT:
+            payload['response_type'] = 'token'
+        response = self.client.post('/v1/o/authorize/', data=payload)
+        self.client.logout()
+        if response.status_code != 302:
+            raise Exception(response.context_data)
+        self.assertEqual(response.status_code, 302)
+        # now extract the authorization code and use it to request an access_token
+        if application.authorization_grant_type == Application.GRANT_IMPLICIT:
+            fragment = parse_qs(urlparse(response['Location']).fragment)
+            tkn = fragment.pop('access_token')[0]
+        else:
+            query_dict = parse_qs(urlparse(response['Location']).query)
+            authorization_code = query_dict.pop('code')
+            token_request_data = {
+                'grant_type': 'authorization_code',
+                'code': authorization_code,
+                'redirect_uri': application.redirect_uris,
+                'client_id': application.client_id,
+                'client_secret': application.client_secret,
+            }
+
+            response = self.client.post('/v1/o/token/', data=token_request_data)
+            self.assertEqual(response.status_code, 200)
+            # Now we have a token and refresh token
+            tkn = response.json()['access_token']
+
+        t = AccessToken.objects.get(token=tkn)
+        return t
+
+    def test_fhir_request_has_header(self):
+        # create an app for a user and obtain a token
+        anna = self._create_user('anna', '123456', fhir_id='19990000000002')
+        capability_a = self._create_capability('token_management', [['DELETE', r'/v1/o/tokens/\d+/']], default=False)
+        application = self._create_application(
+            'an app', grant_type=Application.GRANT_AUTHORIZATION_CODE,
+            redirect_uris='http://example.it')
+        application.scope.add(capability_a)
+        tkn = self._create_test_token(anna, application)
+        # check access token
+        self.assertTrue(DataAccessGrant.objects.filter(
+            beneficiary=anna,
+            application=application,
+        ).exists())
+        # request fhir resource and verify header presents
+        # this will signal reveiver - fetching_data - where header is asserted
+        # Post Django 2.2:  An OSError exception is expected when trying to reach the
+        #                   backend FHIR server and proves authentication worked.
+        with self.assertRaisesRegexp(OSError, "Could not find the TLS certificate file"):
+            self.client.get('/v1/fhir/Patient',
+                            HTTP_AUTHORIZATION="Bearer " + tkn.token)
```

Comment:
@JFU-GIT This works, but has been bothering me :-)  In the BB2-181 PR had started using the HTTMock library for tests in this area.
I'm thinking of updating the 6 other locations using this method with HTTmock.

No need to change in this PR. We can change it as part of BB2-181 or future work.

Reply:
cool